.github/workflows/pr.yaml

```diff
@@ -29,6 +29,8 @@ jobs:
         run: cargo +stable install cargo-hack --locked
       - name: cargo clippy
         run: cargo hack clippy --each-feature --no-dev-deps
```

Comment:
seems like `--tests` is incompatible with `--no-dev-deps` so I ran the clippy tests manually on just the tests for `p slatedb`

Reply:
Should we include `--each-feature` here still? Otherwise, I think non-default feature flagged test code might not get checked?

Reply:
Sorry, I meant in the lines below 👇

Reply:
Done, needed to fix some stuff on `sst.rs` too

Reply:
Hmm looks like clippy actually takes a significant amount of time... do we think it's worth it?

With `--each-feature`
![image](https://github.com/user-attachments/assets/7adf0df5-faf1-453c-bad9-0527e3ab5010)

With just `-p slatedb`
![image](https://github.com/user-attachments/assets/0b3e01f0-39fb-46e0-8959-0c3c260cab23)

And baseline:

![image](https://github.com/user-attachments/assets/f9e53f91-ceec-4398-820a-5b2bde88bdff)

Reply:
Hmm.. that's annoying, but I guess it makes sense... Lots of permutations 🤪 

Given that this is for tests, let's do `-p slatedb`.